sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/KafkaIOTest.java

```diff
@@ -691,6 +710,24 @@ public void testSinkDisplayData() {
     assertThat(displayData, hasDisplayItem("retries", 3));
   }
 
+  @Test
+  public void testInferKeyCoder() {
+    CoderRegistry registry = new CoderRegistry();
+    registry.registerStandardCoders();
+
+    assertTrue(KafkaIO.getOrInferCoder(registry, LongDeserializer.class, null)
```

Comment:
I think this is fine. It will not silently cause a user's pipeline to differently interpret the data, right? cause we don't automatically infer deserializers from coders, only vice versa.